### ISSUE TITLE
using with a single websocket connection for the whole app

### ISSUE THREAD

**Issue by chee** [serenity-kit/secsync#114]
hey, nik! fabulous library, i really need it. 

this issue is more of a chat than an issue.

i'm wondering if there's any ideas for a strategy for using this with automerge-repo, where you have a single websocket connection for the current tab (or browser, if you use a worker) rather than the yjs style of one connection per doc.

i started hacking something together, but i tripped up on the requirement for websocketEndpoint on a per-doc basis. i'm imagining a world where i could basically wrap an automerge provider (like [herb's local-first/auth](https://github.com/local-first-web/auth/blob/main/demos/automerge-repo-todos/src/util/initializeAuthRepo.ts#L14-L19)) and then every doc fetched and used through that connection would be encoded and deciphered via the keys registered at init

**Comment by nikgraf** [comment]
hey @chee, I'm thinking exactly about this problem a lot these days

Automerge-repo has the concept of a https://github.com/automerge/automerge-repo/blob/06126099322474bb8c6251b58b1ee5e63c174f06/packages/automerge-repo/src/synchronizer/CollectionSynchronizer.ts
This makes it really nice to use, but also wondering how this can scale beyond a single server without a different implementation. Need to investigate more.

One thing I was thinking of is to create a collection concept, a HTTP API and a sync mechanism that does scheduled checks if a sync is needed. My thinking was that the currently open document could have an active websocket connection and all other documents could sync in the background.
This probably also could work happen as a background task for Mobile apps?

I'm curious if this would help you with your project. What's your use-case?

**Comment by nikgraf** [comment]
yeah, I'm talking to the automerge-repo people and I'm contributing and reviewing ideas. Looks promising :)

Thanks for the insights! Might be a good test for future versions :)

**Comment by chee** [comment]
> yeah, I'm talking to the automerge-repo people and I'm contributing and reviewing ideas. Looks promising, but still early :)
> 
> Thanks for the insights! Might be a good test for future versions :)

extremely exciting! i'll work on the other 14,000 items on my todo list in the meantime ❤️
